aeneas/src/ir/VariantSolver.v3

```diff
@@ -17,7 +17,10 @@ def getScalarFromSet(s: Scalar.set) -> Scalar {
 	return Scalar.B32; // should be unreachable
 }
 def getTypeFromScalarSet(s: Scalar.set, usePacking: bool) -> Type {
-	for (i in s) return if(usePacking, IntRep.getType(i), i.ty);
+	for (i in s) {
+		if (i.tag >= Scalar.R32.tag) return AnyRef.TYPE;
```

Comment:
Checking the tag here is somewhat brittle; I think a `match` would be best here.